chat_agent.py

```
@@ -39,21 +41,49 @@ def wikisearch_engine(index):
             metadata=ToolMetadata(
                 name="Wikipedia",
-                description="Useful for performing searches on the wikipedia knowledgebase"
+                description="Useful for performing searches on the wikipedia knowledgebase",
             ),
         )
     ]
 
-    openai.api_key = get_apikey()
+    openai.api_key = get_apikey('openai').get('api_key')
     llm = OpenAI(model=MODEL)
+    def add_page(page_title: str) -> str:
```

Flake8 (7.2.0)

[error] expected 1 blank line before a nested definition, found 0

(E306)